rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -2,5 +2,1349 @@
+/// Represents the trust settings for an endpoint.
+pub struct TrustSettings {
+    /// The 'issuerList' Field.
+    pub issuer_list: Option<String>,
+    /// The 'trustList' Field.
+    pub trust_list: Option<String>,
+}
+
+#[derive(Debug, Clone, Default)]
+/// Represents the authentication method for an endpoint.
+pub enum Authentication {
+    #[default]
+    /// Represents anonymous authentication.
+    Anonymous,
+    /// Represents authentication using a certificate.
+    Certificate {
+        /// The 'certificateSecretName' Field.
+        certificate_secret_name: String,
+    },
+    /// Represents authentication using a username and password.
+    UsernamePassword {
+        /// The 'passwordSecretName' Field.
+        password_secret_name: String,
+        /// The 'usernameSecretName' Field.
+        username_secret_name: String,
+    },
+}
+// ~~ From impls ~~
+impl From<adr_name_gen::Device> for Device {
+    fn from(value: adr_name_gen::Device) -> Self {
+        Device {
+            name: value.name,
+            specification: value.specification.into(),
+            status: value.status.map(DeviceStatus::from),
+        }
+    }
+}
+
+impl From<adr_name_gen::DeviceUpdateEventTelemetry> for Device {
+    fn from(value: adr_name_gen::DeviceUpdateEventTelemetry) -> Self {
+        Device {
+            name: value.device_update_event.device.name,
+            specification: value.device_update_event.device.specification.into(),
+            status: value
+                .device_update_event
+                .device
+                .status
+                .map(DeviceStatus::from),
+        }
+    }
+}
+
+impl From<adr_name_gen::DeviceSpecificationSchema> for DeviceSpecification {
+    fn from(value: adr_name_gen::DeviceSpecificationSchema) -> Self {
+        DeviceSpecification {
+            attributes: value.attributes.unwrap_or_default(),
+            discovered_device_ref: value.discovered_device_ref,
+            enabled: value.enabled,
+            endpoints: value
+                .endpoints
+                .map(DeviceEndpoints::from)
+                .unwrap_or_default(),
+            external_device_id: value.external_device_id,
+            last_transition_time: value.last_transition_time,
+            manufacturer: value.manufacturer,
+            model: value.model,
+            operating_system: value.operating_system,
+            operating_system_version: value.operating_system_version,
+            uuid: value.uuid,
+            version: value.version,
+        }
+    }
+}
+
+impl From<adr_name_gen::DeviceEndpointSchema> for DeviceEndpoints {
+    fn from(value: adr_name_gen::DeviceEndpointSchema) -> Self {
+        let inbound = match value.inbound {
+            Some(inbound_endpoints) => inbound_endpoints
+                .into_iter()
+                .map(|(k, v)| (k, v.into()))
+                .collect(),
+            None => HashMap::new(),
+        };
+        let outbound_assigned;
```

Comment:
Could you initialize them as empty hashmaps here?

```suggestion
        let mut outbound_assigned = HashMap::new();
        let mut outbound_unassigned = HashMap::new();
```

In the `map` after the `into_iter` you can do

`.map(|(k, v)| outbound_assigned.insert(k, OutboundEndpoint::from(v))))`